src/vs/base/browser/markdownRenderer.ts

```diff
@@ -218,7 +218,7 @@ export function renderMarkdown(markdown: IMarkdownString, options: MarkdownRende
 	// We always pass the output through insane after this so that we don't rely on
 	// marked for sanitization.
 	markedOptions.sanitizer = (html: string): string => {
-		const match = markdown.isTrusted ? html.match(/^(<span[^<]+>)|(<\/\s*span>)$/) : undefined;
+		const match = markdown.isTrusted ? html.match(/^(<span[^>]+>)|(<\/\s*span>)$/) : undefined;
```

Comment:
I'm not sure that `[^>]` is correct. We still want to prevent html tags from beginning.  `[^<>]` would probably be the best.

Reply:
> An HTML tag can’t begin inside the attributes of another HTML tag.

I wrote this kind of hurriedly. I think you're right in that this regex is not a *perfect* way to do what's being attempted here, which is to allow `<span class="codicon codicon-something" style="color:#ffff">` defined later in the file:
https://github.com/microsoft/vscode/pull/111258/files#diff-a4165063d59c45ee5b74f5e37e26420a00e6832c8c40233fc5c77dac98c6fa48R307

It's a bigger fix than just a typo, but we might want to be safe by going so far as to allowlist this specific set of inputs, since the regexes are already there in part:

```javascript
/^(?:<span (?:(?:class="codicon codicon-[a-z\-]+(?:codicon-animation-[a-z\-]+)?")?|(?:style="(?:color\:#[0-9a-fA-F]+;|background-color\:#[0-9a-fA-F]+;)"))>|^<\/\s*span>)$/g
```

refiddle: http://refiddle.com/refiddles/5fbe763e75622d16b3130000
rail diagram: [![rail diagram](https://user-images.githubusercontent.com/4625198/100248283-368d4e80-2f33-11eb-932c-0095d68fc7e4.png)](https://regexper.com/#%2F%5E%28%3F%3A%3Cspan%20%28%3F%3A%28%3F%3Aclass%3D%22codicon%20codicon-%5Ba-z%5C-%5D%2B%28%3F%3Acodicon-animation-%5Ba-z%5C-%5D%2B%29%3F%22%29%3F%7C%28%3F%3A%28%3F%3Acolor%5C%3A%23%5B0-9a-fA-F%5D%2B%3B%29%3F%28%3F%3Abackground-color%5C%3A%23%5B0-9a-fA-F%5D%2B%3B%29%3F%22%29%29%3E%7C%5E%3C%5C%2F%5Cs*span%3E%29%24%2F)

Reply:
Instead of blocking a specific character, can we only allow specific characters? Something like:

```
<span( [a-z\-'"#0-9]*)?>
```

Reply:
It *might* be possible to confuse that regex into showing something it's not meant to by ending each pass with a trailing quote:

```HTML
<span "> "this text may not be interpreted as HTML, but is parsed as HTML now"
```

Reply:
It should be safe if you require it to match the entire string